### PR TITLE
chore(perf): Make data repo creation async in Compliance

### PR DESCRIPTION
**Pull request by connorgorman**
## Description

This makes the trigger scan call return quickly while also adding a field to the API that allows for the UI to check the latest run statuses. This is important because it allows for the UI to navigate back to the page to get the most up to date run statuses including if they are in progress.

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

### Here I tell how I validated my change

Will test

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: central/compliance/manager/manager_impl.go [hunk 1]
```diff
@@ -90,29 +96,41 @@ func newManager(standardsRegistry *standards.Registry, complianceOperatorManager
...
 func (m *manager) createDomain(ctx context.Context, clusterID string) (framework.ComplianceDomain, error) {
+	t := time.Now()
 	cluster, ok, err := m.clusterStore.GetCluster(ctx, clusterID)
 	if err == nil && !ok {
 		err = errors.New("cluster not found")
 	}
 	if err != nil {
 		return nil, errors.Wrapf(err, "could not get cluster with ID %q", clusterID)
 	}
+	log.Infof("Took %d milliseconds to get cluster", time.Since(t).Milliseconds())
```
Comment (dashrews78) [inline central/compliance/manager/manager_impl.go:107]: Should these be metrics instead or are they debugging artifacts?

**Review by dvail — APPROVED** [review]
Approving from the UI side based on Van's approval [here](https://github.com/stackrox/stackrox/pull/9363) and my testing during development.

**Comment by connorgorman** [comment]
@dashrews78 This is ready to go now

**Review by dashrews78 — APPROVED** [review]
Looks good.  Did leave a comment on whether all those logs should be metrics instead.  But that could be follow on work if we want to do it.

File: central/compliance/manager/manager_impl.go (hunk 1 above)
Comment (connorgorman) [reply central/compliance/manager/manager_impl.go]: Debugging, I removed them

**Pull request merged**